src/component.rs

```diff
@@ -105,6 +105,18 @@ impl<'a> Component<'a> {
     }
 }
 
+impl<'a> From<&'a Package> for Component<'a> {
```

Comment:
This was causing the biggest compilation issue because of this method:

```
    /// Amend the BOM with the specified crates as `components` entries.
    pub fn with_dependencies(mut self, packages: impl IntoIterator<Item = &'a Package>) -> Self {
        self.components
            .extend(packages.into_iter().map(Component::from));
        self
    }
```

Because there wasn't a `from` for `Component`, it was maaaaad